src/Aspire.Dashboard/Model/ResourceEndpointHelpers.cs

```diff
@@ -36,7 +39,8 @@ public static List<DisplayedEndpoint> GetEndpoints(ResourceViewModel resource, b
         // - other urls
         // - endpoint name
         var orderedEndpoints = endpoints
-            .OrderByDescending(e => e.Url?.StartsWith("https") == true)
+            .OrderByDescending(e => e.SortOrder ?? 0)
```

Comment:
Unit test this works as expected. There should be a test that combines all sort order possibilities: 0, null, positive number, negative number.